modules/subscriptions.php

```diff
@@ -708,12 +715,12 @@ function set_cookies( $comments = true, $posts = true ) {
 		 */
 		$cookie_domain   = apply_filters( 'jetpack_comment_cookie_domain', COOKIE_DOMAIN );
 
-		if ( $comments )
-			setcookie( 'jetpack_comments_subscribe_' . self::$hash, $post->ID, time() + $cookie_lifetime, $cookie_path, $cookie_domain );
+		if ( $subscribe_to_post && $post_id >= 0 )
```

Comment:
While we're touching these lines, would be nice to upgrade them to WP coding standards (adding braces), but not a blocker.